draft-ietf-ppm-dap.md

```diff
@@ -1200,34 +1182,120 @@ enum {
+} PrepareStepState;
+
+struct {
+  PrepareStepState prepare_step_state;
+  select (PrepareStep.prepare_step_state) {
+    case initialize:
+      ReportShare report_share;
+      opaque payload<0..2^32-1>;
+    case continue:
+      ReportId report_id;
+      opaque payload<0..2^32-1>;
+    case finished:
+      ReportId report_id;
+    case reject:
+      ReportId report_id;
+      ReportShareError report_share_error;
+  };
+} PrepareStep;
+~~~
+
+### Aggregate Initialization {#agg-init}
+
+The Leader begins an aggregation job by choosing a set of candidate reports that
+pertain to the same DAP task and a unique job ID. The job ID is a 16-byte value,
+structured as follows:
+
+~~~
+opaque AggregationJobID[16];
+~~~
+
+The Leader can run this process for many sets of candidate reports in parallel
+as needed. After choosing a set of candidates, the Leader begins aggregation by
+splitting each report into report shares, one for each Aggregator. The Leader
+and Helper then run the aggregate initialization flow to accomplish two tasks:
+
+1. Recover and determine which input report shares are valid.
+1. For each valid report share, initialize the VDAF preparation process (see
+   {{Section 5.2 of !VDAF}}).
+
 The Leader and Helper initialization behavior is detailed below.
 
 #### Leader Initialization {#leader-init}
 
 The Leader begins the aggregate initialization phase with the set of candidate
-report shares as follows:
+reports as follows:
 
 1. Generate a fresh AggregationJobID. The ID value MUST be unique within the
    scope of the corresponding DAP task.
 1. Decrypt the input share for each report share as described in
    {{input-share-decryption}}.
 1. Check that the resulting input share is valid as described in
    {{input-share-validation}}.
-1. Initialize VDAF preparation as described in {{input-share-prep}}.
 
-If any step invalidates the report share, the Leader removes the report share
-from the set of candidate reports. Once the leader has initialized this state
-for all valid candidate report shares, it creates an `AggregationJobInitReq`
-message for each Helper to initialize the preparation of this candidate set. The
-`AggregationJobInitReq` message is structured as follows:
+If any step invalidates the report, the Leader rejects the report and removes
+it from the set of candidate reports.
+
+Next, for each report the Leader executes the following procedure:
+
+~~~
+state = VDAF.ping_pong_start(vdaf_verify_key,
+                             True,
+                             agg_param,
+                             report_id,
+                             public_share,
+                             plaintext_input_share.payload)
+if state != Rejected():
+  (state, outbound) = VDAF.ping_pong_req(agg_param, state, None)
+~~~
+
+where:
+
+* `vdaf_verify_key` is the VDAF verification key for the task
+* `agg_param` is the VDAF aggregation parameter provided by the Collector (see
+  {{collect-flow}})
+* `report_id` is the report ID
```

Comment:
nit: I think we should clarify that the report ID is used as the nonce.

Reply:
Bump, still no mention of the VDAF nonce in this section at dcced627426a8db038b99c1c9918fe97de93299e

Reply:
Looking to https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/commit/dcced627426a8db038b99c1c9918fe97de93299e, the relevant text now says: "`report_id` is the report ID, used as a nonce"

Reply:
Oops, never mind, I was holding it wrong.